src/stream-tcp.c

```diff
@@ -5799,6 +5801,14 @@ int StreamTcpBypassEnabled(void)
     return stream_config.bypass;
 }
 
+void StreamTcpSetReassemblyDepth(TcpSession *ssn, uint32_t size)
+{
+    ssn->server.reassembly_depth = size;
```

Comment:
It seems that none of this work takes that future work into account. None of the API calls accept a direction flag for example. So it seems it's just about StreamTcpReassembleCheckDepth not having a ssn ptr.